editor/src/components/canvas/dom-walker.ts

```diff
@@ -346,23 +367,23 @@ export function useDomWalker(props: CanvasContainerProps): React.Ref<HTMLDivElem
 
 function collectMetadata(
   element: HTMLElement,
+  uidAttribute: string | null,
   instancePath: InstancePath,
   parentPoint: CanvasPoint,
   children: InstancePath[],
   scale: number,
   containerRectLazy: () => CanvasRectangle,
   invalidatedPathsForStylesheetCacheRef: React.MutableRefObject<Set<string>>,
   selectedViews: Array<TemplatePath>,
-) {
+): ElementInstanceMetadata {
   const globalFrame = globalFrameForElement(element, scale, containerRectLazy)
   const localFrame = localRectangle(Utils.offsetRect(globalFrame, Utils.negate(parentPoint)))
 
-  const uidAttribute = getDOMAttribute(element, UTOPIA_UID_KEY)
   const originalUIDAttribute = getDOMAttribute(element, UTOPIA_ORIGINAL_ID_KEY)
   const labelAttribute = getDOMAttribute(element, UTOPIA_LABEL_KEY)
   let elementProps: any = {}
   if (uidAttribute != null) {
-    elementProps[UTOPIA_UID_KEY] = uidAttribute
+    elementProps[UTOPIA_UIDS_KEY] = uidAttribute // TODO Balazs we are making a fake prop with a single UID instead of the UID array – maybe this means changes to mergeComponentMetadata
```

Comment:
This (and the other instance later down with a similar comment) are also going to be binned in my "other PR"